Теперь стеклянные бутылки разбиваются об голову

mofkkol opened the pull request:
<!-- Рекомендации: https://docs.spacestation14.io/en/getting-started/pr-guideline -->

## Описание PR
<!-- Что вы изменили? -->
Стеклянные бутылки теперь могут ломаться об голову людей, нанося урон в 10 ушибов.
## Почему / Баланс
<!-- Обсудите, как это повлияет на баланс игры или объясните, почему это было изменено. Укажите ссылки на соответствующие обсуждения или issue. -->
Принятая предложка: https://discord.com/channels/1066727245806325872/1391533230099927041
## Технические детали
<!-- Краткое описание изменений в коде для облегчения проверки. -->
Сделано на основе акустической гитары.
## Медиа
<!-- Прикрепите медиафайлы, если PR вносит изменения в игру (одежда, предметы, механики и т.д.).
Небольшие исправления/рефакторинг освобождаются от этого требования. -->

https://github.com/user-attachments/assets/cbe03e2e-0d43-47c1-8734-0c9869ddf40a


## Требования
<!-- Подтвердите следующее, поставив X в скобках [X]: -->
- [x] Я прочитал(а) и следую [Рекомендациям по оформлению Pull Request и Changelog](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] Я добавил(а) медиафайлы к этому PR или он не требует демонстрации в игре.

## Согласие с условиями
- [x] Я согласен с условиями [LICENSE](../LICENSE.md) и [CLA](../CLA.md).

<!-- Вы должны понимать, что несоблюдение вышеуказанного может привести к закрытию вашего PR по усмотрению сопровождающего -->

**Список изменений**
<!-- Добавьте запись в Changelog, чтобы игроки знали о новых функциях или изменениях, которые могут повлиять на игровой процесс.
Убедитесь, что вы прочитали рекомендации и вынесли этот шаблон Changelog из блока комментариев, чтобы он отображался.
Changelog должен иметь символ :cl:, чтобы бот распознал изменения и добавил их в список изменений игры. -->

:cl: mofkkol, reartint
- wl-add: Теперь стеклянные бутылки разбиваются об голову.


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Many new empty drink containers added (glass bottles and large cartons).
  * Trash drink items now carry forensic traces for investigation.
  * Bottles can be used as melee weapons with noticeable blunt damage and animations.

* **Gameplay Changes**
  * Glass bottles are now explicitly destructible (often break in a single hit), spill contents, produce shatter effects, and leave debris.
  * Forensics and destruction flows around bottles have been expanded/adjusted.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

mofkkol committed:
renga

mofkkol committed:
start

mofkkol committed:
fixes

mofkkol committed:
fix2

cfif126 commented on Resources/Prototypes/Entities/Objects/Consumable/Drinks/drinks_bottles.yml:
```diff
@@ -127,13 +137,20 @@
+    wideAnimationRotation: 45
+    damage:
+      types:
+        Blunt: 5
```
Считаю 10 ушибов золотой серединой

mofkkol committed:
Update Resources/Prototypes/Entities/Objects/Consumable/Drinks/drinks_bottles.yml

Co-authored-by: tau <64920518+tau27@users.noreply.github.com>

mofkkol committed:
Update Resources/Prototypes/Entities/Objects/Consumable/Drinks/drinks_bottles.yml

Co-authored-by: tau <64920518+tau27@users.noreply.github.com>

mofkkol committed:
Update Resources/Prototypes/Entities/Objects/Consumable/Drinks/drinks_bottles.yml

Co-authored-by: tau <64920518+tau27@users.noreply.github.com>

mofkkol committed:
Update Resources/Prototypes/Entities/Objects/Consumable/Drinks/drinks_bottles.yml

Co-authored-by: tau <64920518+tau27@users.noreply.github.com>

mofkkol committed:
Update Resources/Prototypes/Entities/Objects/Consumable/Drinks/trash_drinks.yml

Co-authored-by: tau <64920518+tau27@users.noreply.github.com>

mofkkol committed:
Update Resources/Prototypes/Entities/Objects/Consumable/Drinks/trash_drinks.yml

Co-authored-by: tau <64920518+tau27@users.noreply.github.com>

mofkkol reviewed (commented):
@tau27 Исправлено

tau27 commented on Resources/Prototypes/Entities/Objects/Consumable/Drinks/trash_drinks.yml:
```diff
@@ -66,17 +65,26 @@
+    damage:
+      types:
+        Blunt: 15
+  - type: DamageOtherOnHit
+    damage:
+      types:
+        Blunt: 4
```
М, точно. Тогда тут всё хорошо, не меняем.

tau27 commented on Resources/Prototypes/Entities/Objects/Consumable/Drinks/drinks_bottles.yml:
```diff
@@ -110,6 +110,16 @@
+#WL-Changes-Start
+  - type: Damageable # Smash it on a head of your enemies! Does 10 damage a hit, but breaks after 1 hit.
+    damageContainer: Inorganic
   - type: Destructible
```
Компонент Destructible уже был в прототипе, его бы вынести за пределы WL-Changes

tau27 commented on Resources/Prototypes/Entities/Objects/Consumable/Drinks/drinks_bottles.yml:
```diff
@@ -150,7 +163,15 @@
+#WL-Changes-Start
+  - type: Damageable # Smash it on a head of your enemies! Does 10 damage a hit, but breaks after 1 hit.
     damageContainer: Inorganic
   - type: Destructible
```
Аналогично случаю выше

tau27 commented on Resources/Prototypes/Entities/Objects/Consumable/Drinks/trash_drinks.yml:
```diff
@@ -66,17 +65,22 @@
+    damage:
+      types:
+        Blunt: 15
+  - type: DamageOtherOnHit
+    damage:
+      types:
+        Blunt: 4
```
Этот компонент был в изначальном файле, так что его бы тоже вынести за WL-Changes

tau27 commented on Resources/Prototypes/Entities/Objects/Consumable/Drinks/trash_drinks.yml:
```diff
@@ -90,6 +94,40 @@
     solutions:
       drink:
         maxVol: 50
+#WL-Changes-Start
```
DrinkBottleBaseSmallEmpty наследуется от DrinkBottleBaseEmpty, так что эти изменения являются лишним дупликатом, убрать бы

tau27 reviewed (changes requested):
Учитывая, что тут однотипные изменения повторяются по несколько раз, логично будет вынести их в отдельный абстрактный прототип (его вынести в папку _WL) и просто добавить бутылкам наследование от него.

mofkkol committed:
fixes

# Conflicts:
#	Resources/Prototypes/Entities/Objects/Consumable/Drinks/trash_drinks.yml

JrInventor05 commented:
а почему бы не сделать автоподбор в активную руку осколка бутылки? Мы же разбив бутылку не выронили её, но тут как бы роняем...
